### PR TITLE
Gear be/fdr 50 be gateway checkin

### PR DESCRIPTION
**Pull request by macgeargear**
## Change made

- [x]  New features
- [ ]  Bug fixes
- [ ]  Breaking changes
- [ ]  Refactor
## Describe what you have done
- create gate way
### New Features
- 
### Fix
-

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -34,16 +36,29 @@ func main() {
...
+
 	r := router.New(conf, corsHandler, authMiddleware)
 
 	r.V1Get("/auth/google-url", authHdr.GetGoogleLoginUrl)
 	r.V1Post("/auth/verify-google", authHdr.VerifyGoogleLogin)
 	r.V1Post("/auth/test", authHdr.Test)
 
+	r.V1Post("/checkin", checkinHdr.Create)
+	r.V1Post("/checkin/userId/", checkinHdr.FindByUserID)
+	r.V1Post("/checkin/email/", checkinHdr.FindByEmail)
```
Comment (bookpanda) [inline cmd/main.go:60]: let's do it this way; it's more conventional
/checkin/:userId
/chcekin/email/:email

File: internal/checkin/checkin.handler.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+		h.log.Named("FindByEmail").Error("Bind: failed to bind request body", zap.Error(err))
+		c.BadRequestError(err.Error())
+		return
+	}
+
+	if errorList := h.validate.Validate(body); errorList != nil {
+		h.log.Named("FindByEmail").Error("Validate: ", zap.Strings("errorList", errorList))
+		c.BadRequestError(strings.Join(errorList, ", "))
+		return
+	}
```
Comment (bookpanda) [inline internal/checkin/checkin.handler.go:82]: now email comes from url params, no need for body, validation

File: internal/checkin/checkin.handler.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+		h.log.Named("FindByUserID").Error("Bind: failed to bind request body", zap.Error(err))
+		c.BadRequestError(err.Error())
+		return
+	}
+
+	if errorList := h.validate.Validate(body); errorList != nil {
+		h.log.Named("FindByUserID").Error("Validate: ", zap.Strings("errorList", errorList))
+		c.BadRequestError(strings.Join(errorList, ", "))
+		return
+	}
```
Comment (bookpanda) [inline internal/checkin/checkin.handler.go:112]: same here

**Review by bookpanda — CHANGES_REQUESTED** [review]
only need some changes to
/checkin/:userId
/chcekin/email/:email

File: cmd/main.go (hunk 1 above)
Comment (macgeargear) [reply cmd/main.go:60]: 🫡

**Review by bookpanda — APPROVED** [review]
lgtm

**Pull request merged**